dkpro-core-io-brat-asl/src/main/java/de/tudarmstadt/ukp/dkpro/core/io/brat/BratWriter.java

```diff
@@ -306,36 +336,41 @@ private void writeAnnotations(JCas aJCas)
 
         // Go through all the annotations but only handle the ones that have no references to
         // other annotations.
+        for (String currentType: spanTypes.keySet()){
+             for (FeatureStructure fs : selectAll(aJCas)) {
+                String typeName=fs.getType().getName();
+                String superType = fs.getCAS().getTypeSystem().getParent(fs.getType()).getName();
```

Comment:
This code seems to take into account only the type and the direct supertype of the FS. When taking into account the direct supertype, it seems sensible to actually check the entire inheritance hierarchy, no? `org.apache.uima.cas.TypeSystem.subsumes(Type, Type)` should be useful here.

Comment applies also to other changes further down.

Reply:
I just used the same way it was used somewhere else in the module. I understood that it was a method to specify the parent type instead of the all the subsumed types.